cloudwatch-event.tf

```diff
@@ -0,0 +1,115 @@
+data "archive_file" "apex_restart" {
+  type        = "zip"
+  source_dir  = "../../functions/"
+  output_path = "../../functions/apex_restart.zip"
+}
+
+resource "aws_lambda_function" "apex_restart" {
```

Comment:
All these resources need to be conditional based on a var.  Maybe consider putting this in an internal module so all the resources can be `count = var.weekly_restart_enabled ? 1 : 0`'d with a single line (as opposed to having to jam that on every resource.  As long as you can keep the vars passed to the internal module to a minimum, i think that makes sense.  If you'd need to pass a dozen vars, maybe it's not worth it?